src/repository.py

```diff
@@ -34,6 +41,47 @@
 CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
 
 
+class DiffSummary(NamedTuple):
+    is_dirty: bool
+    new: Sequence[str]
+    removed: Sequence[str]
+    modified: Sequence[str]
+
+    @classmethod
+    def from_raw_diff(cls, diffs: Sequence[Diff]):
+        new_files = {diff.a_path for diff in diffs if diff.new_file}
+        removed_files = {diff.a_path for diff in diffs if diff.deleted_file}
+        modified_files = {diff.a_path for diff in diffs if diff.renamed_file}.union(
+            diff.a_path for diff in diffs if diff.change_type == "M"
+        )
+
+        return DiffSummary(
+            is_dirty=len(diffs) > 0,
+            new=list(new_files),
+            removed=list(removed_files),
+            modified=list(modified_files)
+        )
+
+    def __add__(self, other: 'DiffSummary'):
+        if not isinstance(other, DiffSummary):
+            raise ValueError("I can add only DiffSummary classes")
+
+        return DiffSummary(
+            is_dirty=self.is_dirty or other.is_dirty,
+            new=list(set(self.new).union(other.new)),
+            removed=list(set(self.removed).union(other.removed)),
+            modified=list(set(self.modified).union(other.modified)),
+        )
+
+    def __str__(self) -> str:
+        modified_str = [f"modified: {','.join(self.modified)}"] if len(self.modified) > 0 else []
+        new_str = [f"new: {','.join(self.new)}"] if len(self.new) > 0 else []
+        removed_str = [f"removed: {','.join(self.removed)}"] if len(self.removed) > 0 else []
+        return " // ".join(
+            modified_str + new_str + removed_str
```

Comment:
f-string?

Reply:
Uhm, I'm not sure how I could get the same result with f-string. I mean here I want to join a list of elements which can be of variable length (size of the list should be between 0 and 3 elements). Do you have a particular form in mind?

Reply:
oh I misread, I though that `modified_str` was a string. Perhaps the variable name can be clarified, e.g., `modified_files`?

Reply:
It is also possible to use `itertools.chain` to avoid needing to create new lists to be able to iterate over all of them, a slight increase in efficiency